src/Tasks/AssemblyDependency/ResolveAssemblyReference.cs

```diff
@@ -2448,17 +2444,17 @@ private AssemblyNameExtension[] GetDependencies(Reference resolvedReference, Fil
         /// <summary>
         /// Combines two DependentAssembly arrays into one.
         /// </summary>
-        private static DependentAssembly[] CombineRemappedAssemblies(DependentAssembly[] first, DependentAssembly[] second)
+        private static List<DependentAssembly> CombineRemappedAssemblies(List<DependentAssembly> first, List<DependentAssembly> second)
         {
             if (first == null)
                 return second;
```

Comment:
Not safe in the sense that you mean; but safe in the sense that ReferenceTable.ComputeClosure(), the only consumer of the resulting list, treats the list as immutable. Updated in next push to make ReferenceTable use IEnumerable instead of List to ensure it's immutable.

Reply:
I realized I missed the gist of what you were saying. Additional push changes the signature here to use IReadOnlyCollection to avoid unintended mutability.